src/hooks/useRawSearchParams.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+
+type UpdateSearchParamsOptions = {
+  method?: 'replace' | 'push';
+};
+
+type UpdateSearchParams = (
+  newSearchParams: URLSearchParams,
+  options?: UpdateSearchParamsOptions,
+) => void;
+
+type RawSearchParams = [URLSearchParams, UpdateSearchParams];
+
+/**
+ * This hook provides an `updateSearchParams` function that can update the
+ * search parameters _without_ creating a re-render on the router level.
+ */
+const useRawSearchParams = (): RawSearchParams => {
+  // Memoize the current search parameters to avoid unnecessary recalculations
+  const searchParams = useMemo(() => {
```

Comment:
I used `useMemo` here but probably it's a bit of an overkill... It should be quite fast to construct a new `URLSearchParams`. Let me know if you prefer me to remove it instead.